Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Payment.hs

```diff
@@ -224,27 +224,30 @@ notifyAndUpdateInvoiceStatusIfPaymentFailed driverId orderId orderStatus eventNa
         QDF.updateAutoPayToManual invoice'.driverFeeId
         QDF.updateAutopayPaymentStageById (Just EXECUTION_FAILED) invoice'.driverFeeId
       Nothing -> pure ()
-    nofityPaymentFailureIfNotNotified paymentMode
+    nofityUltimatePaymentFailureIfNotNotifiedWithoutLimitAndError paymentMode
 
   when (notifyFailure && isJust mbBankErrorCode) $
```

Comment:
when both of these `notifyFailure, updateFailure` are true then we will end up sending notification twice.

Reply:
updated the conditions